scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -65,5 +65,9 @@ def _logerror(self, failure, request, spider):
 
     def _parse_robots(self, response):
         rp = robotparser.RobotFileParser(response.url)
-        rp.parse(response.body.splitlines())
+        # FIXME: unsure if we should do this here
+        # (or enforce the .encoding attr in `Response`)
+        if not hasattr(response, 'encoding'):
+            response.encoding = 'utf-8'
```

Comment:
Adding an extra fallback here sounds OK to me, for the case server returns some strange headers for /robots.txt request, but the file contents is ok. But it is better not to patch the response. It should use `response.body_as_unicode()` when possible, and fall back to `response.body.decode('...')` when it fails.